ext/opcache/jit/zend_jit_x86.dasc

```diff
@@ -10510,13 +10510,15 @@ static int zend_jit_fetch_obj_read(dasm_State **Dst, const zend_op *opline, cons
 			}
 			|.if X64
 				|	mov CARG1, E_WARNING
-				|	LOAD_ADDR CARG2, "Trying to get property '%s' of non-object"
+				|	LOAD_ADDR CARG2, "Access to property '%s' on %s"
 				|	LOAD_ADDR CARG3, Z_STRVAL_P(member)
+				|	LOAD_ADDR CARG4, "non-object"
```

Comment:
this is a problem :( I tried to retrieve an `IS_*` value from the `op1_info`, but it didn't always work.

Reply:
op1_info are possible types, you need the concrete type here. I'd move this code into a jit helper and then call it with arguments `container` and `Z_STRVAL_P(member)`. The container should be in `FCARG1a` at this point, so you don't actually need to move it anywhere.

Reply:
I managed to at least partially fix the JIT code, but I still get a segfault in some cases. Hopefully, it was ok to remove the architectural condition.

P.S. I found a bit concerning `Attempt to assign property of non-object` warning which is only defined by JIT, but nowhere else.